modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelFactory.java

```diff
@@ -0,0 +1,453 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Implementation of {@link DeployableProjectModelFactory} that uses Maven.
+ *
+ * @param <M> type of the project model.
+ *
+ * @since 4.5
+ */
+public abstract class AbstractMavenDeployableProjectModelFactory<M extends MuleDeployableModel>
+    implements DeployableProjectModelFactory<M> {
+
+  private static final String DEFAULT_PACKAGE_EXPORT = "";
+  private static final String JAVA_EXTENSION = "java";
+  private static final String PACKAGE_SEPARATOR = ".";
+
+  private static final String GROUP_ID = "groupId";
+  private static final String ARTIFACT_ID = "artifactId";
+  private static final String VERSION = "version";
+
+  private static final String CLASS_PATH_SEPARATOR = "/";
+
+  private static final Logger LOGGER = getLogger(AbstractMavenDeployableProjectModelFactory.class);
+
+  private final File projectFolder;
+  private final MavenConfiguration mavenConfiguration;
+  private List<String> exportedPackages = emptyList();
+  private List<String> exportedResources = emptyList();
+
+  public AbstractMavenDeployableProjectModelFactory(File projectFolder, MavenConfiguration mavenConfiguration) {
+    this.projectFolder = projectFolder;
+    this.mavenConfiguration = mavenConfiguration;
+  }
+
+  public AbstractMavenDeployableProjectModelFactory(File projectFolder) {
+    this(projectFolder, getDefaultMavenConfiguration());
+  }
+
+  private static MavenConfiguration getDefaultMavenConfiguration() {
+    final MavenClientProvider mavenClientProvider =
+        discoverProvider(AbstractMavenDeployableProjectModelFactory.class.getClassLoader());
+    final Supplier<File> localMavenRepository =
+        mavenClientProvider.getLocalRepositorySuppliers().environmentMavenRepositorySupplier();
+
+    final SettingsSupplierFactory settingsSupplierFactory = mavenClientProvider.getSettingsSupplierFactory();
+
+    final Optional<File> globalSettings = settingsSupplierFactory.environmentGlobalSettingsSupplier();
+    final Optional<File> userSettings = settingsSupplierFactory.environmentUserSettingsSupplier();
+    final Optional<File> settingsSecurity = settingsSupplierFactory.environmentSettingsSecuritySupplier();
+
+    final MavenConfiguration.MavenConfigurationBuilder mavenConfigurationBuilder = newMavenConfigurationBuilder()
+        .forcePolicyUpdateNever(true)
+        .localMavenRepositoryLocation(localMavenRepository.get());
+
+    globalSettings.ifPresent(mavenConfigurationBuilder::globalSettingsLocation);
+
+    userSettings.ifPresent(mavenConfigurationBuilder::userSettingsLocation);
+
+    settingsSecurity.ifPresent(mavenConfigurationBuilder::settingsSecurityLocation);
+
+    return mavenConfigurationBuilder.build();
+  }
+
+  @Override
+  public DeployableProjectModel<M> createDeployableProjectModel() {
```

Comment:
this methoid is too long, this should be split. Please also add comments explaining the process to improve readability

Reply:
Improved